Taskfile.yml

```diff
@@ -367,6 +413,31 @@ tasks:
     cmds:
       - "{{.SCRIPTS_ROOT}}/deploy_testing_secret.sh aadpodidentity"
 
+  controller:kind-create-with-service-principal:
+    desc: Creates a local kind cluster with ASO using service principal auth.
+    cmds:
+      - task: controller:kind-create
+      - task: controller:install-cert-manager
+      - task: controller:docker-push-local
+      # We need the below to wait until cert-manager is up, otherwise the subsequent installation of webhooks fails. See https://cert-manager.io/next-docs/installation/verify/
+      - "cmctl check api --wait=2m"
+      - task: controller:install
+      - task: controller:make-sp-secret
+      - "kubectl wait --for=condition=ready --timeout=2m pod -n azureserviceoperator-system -l control-plane=controller-manager"
```

Comment:
Does `kubectl wait` return a non-zero exit code if the timeout expires?

Reply:
yes